### ISSUE TITLE
color theme ignored if loaded after colorama

### ISSUE THREAD

**Issue by ricma** [inducer/pudb#381]
I realized that pudb does not display its color theme correctly after some libraries are loaded.

TL;DR: Package [colorama](https://github.com/tartley/colorama) sets some escape sequences (or not?) - not sure whom to ask for a fix, tbh.

Test:
```
python3 -c "from colorama import init as colorinit; colorinit(autoreset=True); import pudb; pudb.set_trace()"
```

Expected result: Pudb opens in color.
Seen output: Pudb opens in black and white.

Alternative test:
```
python3 -c "from colorama import init as colorinit; colorinit(); import pudb; pudb.set_trace()"
```

This one works as expected.

The problem is triggered somewhere [here, ansitowin32.py, line 167](https://github.com/tartley/colorama/blob/master/colorama/ansitowin32.py#L167) and I don't know if colorama is the problem or pudb, so sorry in case this does not belong here, I am happy to open an issue with colorama instead.

Background: I was using a [library](https://github.com/SteveDoyle2/pyNastran) which itself is using [cpylog](https://github.com/SteveDoyle2/cpylog) and this is as far as I was tracking the issue down.

Versions used:
python3: 3.6.9
pudb: 2019.2
colorama: 0.4.3

**Comment by inducer** [comment]
Thanks for the report. Upon poking through the [`colorama` source](https://github.com/tartley/colorama/blob/2b7f7a70fe8233d3ce64854073883bd5a1758e5f/colorama/initialise.py#L37-L43), it seems that their misguided `autoreset` wraps the stdout and stderr streams to send terminal reset codes after every write to the stream. Those are then the streams that pudb/Urwid talks to. If someone hands us a broken stdout, there's little we can do to fix it, short of specifically detecting and undoing the damage they've done. It's really `colorama`'s bug, or really the problem of anyone who uses their misguided `autoreset` feature.

**Comment by ricma** [comment]
Thank you very much for your clarification - I was not able to understand the inner workings to that level.

I will contact @tartley and let him know.

Thanks again! (also for _pudb_ in general - Couldn't live without it and miss it in other programming languages.)

**Comment by tartley** [comment]
I see this, thanks for the ping.

Yep, I concur it sounds like a Colorama bug. I've filed an issue: https://github.com/tartley/colorama/issues/255

Apologies the code is ill-considered. It was a weekend hack I threw together a long time ago, just so that our test output would look pretty even on Windows. Then it rapidly and unexpectedly grew to million downloads a week, and people get mad if we change some of its unintentional behaviors, so we're loathe to make needless changes to it.

But this sounds like a real problem and should be fixed.
